lib/config.py

```diff
@@ -98,56 +99,132 @@ def _setup_config_parser_options(self):
-        self.parser.add_argument('--release-notes-repo-branch',
-                                 help='Branch of the release notes repository '
-                                 'to checkout')
-        self.parser.add_argument('--push-repo-url',
-                                 help='URL of the repository used for pushing')
-        self.parser.add_argument('--push-repo-branch',
-                                 help='Branch of the repository used for '
-                                 'pushing',
-                                 default='master')
-        self.parser.add_argument('--committer-name',
-                                 help='Name used when creating a commit and '
-                                 'bumping spec files')
-        self.parser.add_argument('--committer-email',
-                                 help='Email used when creating a commit and '
-                                 'bumping spec files')
 
+        subparser = self.parser.add_subparsers(
+            dest="subcommand",
+            help="Available subcommands")
+
+        # create the parser for the "build-package" subcommand
+        parser_package = subparser.add_parser('build-package',
+                                              help="Build package")
+        self._add_package_args(parser_package)
+        self._add_build_repo_args(parser_package)
+
+        # create the parser for the "release-notes" subcommand
+        parser_release_notes = subparser.add_parser('release-notes',
+                                                    help="Create release notes")
+        self._add_release_notes_args(parser_release_notes)
+        self._add_push_repo_args(parser_release_notes)
+        self._add_build_repo_args(parser_release_notes)
+
+        # create the parser for the "upgrade-versions" subcommand
+        parser_upgrade_versions = subparser.add_parser('upgrade-versions',
+                                                       help="Upgrade packages "
+                                                       "versions")
+        self._add_push_repo_args(parser_upgrade_versions)
+        self._add_build_repo_args(parser_upgrade_versions)
+
+        # create the parser for the "set-env" subcommand
+        parser_setup_environment = subparser.add_parser('set-env',
+                                                        help="Setup user and "
+                                                        "directory for build "
+                                                        "scripts")
+        self._add_setup_environment_args(parser_setup_environment)
+
+        # create the parser for the "build-iso" subcommand
+        parser_iso = subparser.add_parser('build-iso', help="Build ISO image")
+        self._add_iso_args(parser_iso)
+
+    def _add_build_repo_args(self, subparser):
+        """
+        Configures build repository arguments for subparser object to match the
+            expected configuration.
```

Comment:
If my suggestion above is not taken, I think we should alter these docs to something more descriptive, like: "Args related to building the repo".